aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1308 @@
 
-The preceding sample uses the configuration indexer to read values. To access configuration outside of `Startup`, use the *options pattern*. For more information, see the [Options](xref:fundamentals/configuration/options) topic.
+> [!NOTE]
+> For an example of how to flow command-line arguments from `Program:Main` to the `Startup` class, where they can be used with the Command-line Configuration Provider, see the [1.x sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/configuration/index/samples/1.x). The approach shown in the 1.x sample can also be used in 2.x or later apps.
 
-## XML configuration
+## Command-line Configuration Provider
 
-To work with arrays in XML-formatted configuration sources, provide a `name` index to each element. Use the index to access the values:
+The [CommandLineConfigurationProvider](/dotnet/api/microsoft.extensions.configuration.commandline.commandlineconfigurationprovider) loads configuration from command-line argument key-value pairs at runtime.
 
-```xml
-<wizards>
-  <wizard name="Gandalf">
-    <age>1000</age>
-  </wizard>
-  <wizard name="Harry">
-    <age>17</age>
-  </wizard>
-</wizards>
+::: moniker range=">= aspnetcore-2.0"
+
+To activate command-line configuration, call the [AddCommandLine](/dotnet/api/microsoft.extensions.configuration.commandlineconfigurationextensions) extension method on an instance of [ConfigurationBuilder](/dotnet/api/microsoft.extensions.configuration.configurationbuilder).
+
+`AddCommandLine` is automatically called when you initialize a new [WebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilder) with [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder). For more information, see [Web Host: Set up a host](xref:fundamentals/host/web-host#set-up-a-host).
+
+`CreateDefaultBuilder` also loads:
+
+* Optional configuration from *appsettings.json* and *appsettings.&lt;Environment&gt;.json*.
+* [User secrets (Secret Manager)](xref:security/app-secrets) (in the Development environment).
+* Environment variables.
+
+`CreateDefaultBuilder` adds the Command-line Configuration Provider last. Command-line arguments passed at runtime override configuration set by the other providers.
+
+`CreateDefaultBuilder` acts when the host is constructed. Therefore, command-line configuration activated by `CreateDefaultBuilder` can affect how the host is configured.
+
+For an example of how to flow command-line arguments from `Program:Main` to the `Startup` class so that configuration established in `Startup` can be overridden, see the [1.x sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/configuration/index/samples/1.x). The approach shown in the 1.x sample can also be used in 2.x or later apps.
```

Comment:
Note we no longer recommend establishing the app's config in Startup, that's better done in Main. See ConfigureAppConfiguration

Reply:
*FLU!* ... it got by me when the fever was on me at 101.

I'll update the sample and add a bit about it being for 2.1+.